src/charm.py

```diff
@@ -257,6 +281,14 @@ def _on_event(self, event) -> None:
             ):
                 self._create_default_s3_bucket(s3_wrapper, bucket_name)
             self._update_layer(envs, bucket_name)
+            secrets_context = {
+                "s3_endpoint": f"http://{object_storage_data['service']}.{object_storage_data['namespace']}:{object_storage_data['port']}",  # noqa: E501
+                "s3_type": "s3",
+                "s3_provider": "minio",
+                "access_key": object_storage_data["access-key"],
+                "secret_access_key": object_storage_data["secret-key"],
+            }
+            self._send_manifests(interfaces, secrets_context, SECRETS_TEMPLATE_FOLDER, "secrets")
```

Comment:
This works here, but feels a bit heavy.  We render all manifests with the same context, so it relies on every .j2 file knowing how to avoid name collisions.  It is convenient to implement though.  If you can think of an easy way to avoid this it would be good, otherwise lgtm and we can just fix it if it becomes a problem.

Reply:
Lets resolve this in poddefaults change as its more related to that :)